types/serialization.go

```diff
@@ -40,6 +40,73 @@ func (h *Header) UnmarshalBinary(data []byte) error {
 	return err
 }
 
+// MarshalBinary encodes Header into binary form and returns it.
+func (fp *FraudProof) MarshalBinary() ([]byte, error) {
```

Comment:
Could you also add "round trip" (object -marshal-> bytes -unmarshal-> object)  test in `serialization_test.go` to ensure that serialization works as expected?

Reply:
Added. Notice that removing the files in `proto/pb` and `proto/tendermint` lead to linters and build tests failing now.